frontend/src/js/components/header/DeviceNotifications.tsx

```diff
@@ -11,86 +11,235 @@
+      </div>
+      <div className="flexbox">
+        <Typography variant="caption">{Math.floor((total / limit) * 100)}% used</Typography>
+        {warning && (
+          <>
+            <Typography variant="caption" className="margin-left-x-small margin-right-x-small">
+              •
+            </Typography>{' '}
+            <Typography variant="caption" color="warning">
+              Near limit
+            </Typography>
+          </>
+        )}
+        {error && (
+          <>
+            <Typography variant="caption" className="margin-left-x-small margin-right-x-small">
+              •
+            </Typography>{' '}
+            <Typography variant="caption" color="error">
+              Limit reached
+            </Typography>
+          </>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const DeviceNotifications = ({ className = '', total, pending }) => {
+  const { classes } = useStyles();
+  const navigate = useNavigate();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [tooltip, setTooltip] = useState(false);
+  const handleOpen = (event: MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+    setTooltip(false);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const onManageDeviceLimit = () => {
+    handleClose();
+    navigate('/settings/subscription');
+  };
+  const viewPendingClick = () => {
+    handleClose();
+    navigate('/devices/pending');
+  };
+  const { counts: accepted } = useSelector(getAcceptedDevices);
+  const { isAdmin } = useSelector(getUserRoles);
+
+  const limits = useSelector(getDeviceLimits);
+  const mappedLimits = Object.entries(limits)
+    .filter(([, limit]) => limit > 0)
+    .map(([type, limit]) => ({ type, limit: limit, total: accepted[type] }));
+
+  const severityMap = { 0: 'primary', 1: 'warning', 2: 'error' };
+
+  const maxSeverityIndex = mappedLimits.reduce((maxIndex, { limit, total }) => {
+    if (total >= limit) {
+      return 2;
+    } else if (total / limit > 0.8) {
+      return Math.max(maxIndex, 1);
+    } else {
+      return maxIndex;
+    }
+  }, 0);
+
+  const severity = severityMap[maxSeverityIndex];
+
+  return (
+    <div className={className}>
+      <Tooltip
+        title="View device limits"
+        open={!anchorEl && tooltip}
+        disableFocusListener
+        disableTouchListener
+        disableHoverListener={!!anchorEl}
+        onMouseEnter={() => !anchorEl && setTooltip(true)}
+        onMouseLeave={() => setTooltip(false)}
+      >
+        <div>
+          <Badge color={severity} badgeContent={pending} anchorOrigin={{ horizontal: 'left' }}>
+            <Button
+              startIcon={<DeveloperBoardIcon className="margin-right-x-small" fontSize="small" />}
+              endIcon={<ArrowDropDownIcon fontSize="small" sx={{ color: 'text.secondary' }} className={classes.endIcon} />}
```

Comment:
```suggestion
              endIcon={<ArrowDropDownIcon fontSize="small" className={classes.endIcon} />}
```
considering you have the `endIcon` class defined